cosmo_tester/test_suites/cli/__init__.py

```diff
@@ -57,6 +63,85 @@ def _test_cfy_install(run, example, paths, logger):
     example.check_files()
 
 
+def _set_ssh_in_profile(run, example, paths):
+    run(
+        '{cfy} profiles set --ssh-user {ssh_user} --ssh-key {ssh_key}'.format(
+            cfy=paths['cfy'],
+            ssh_user=example.manager.username,
+            ssh_key=paths['ssh_key'],
+        )
+    )
+
+
+def _test_cfy_logs(run, cli_host, example, paths, logger):
+    _set_ssh_in_profile(run, example, paths)
+
+    linebreak = '\\r\\n' if 'C:\\' in paths['cfy'] else '\n'
+    logger.info('Testing `cfy logs download`')
+    logs_dump_filepath = str(run(
+        '{cfy} logs download '.format(cfy=paths['cfy'])
+    ).stdout).split('archive to:')[1].split(linebreak)[0]
+
+    # Not working on windows!
+    local_logs_dump_filepath = os.path.join(tempfile.mkdtemp(), 'logs.tar')
+    cli_host.get_remote_file(logs_dump_filepath, local_logs_dump_filepath)
+    with closing(tarfile.open(name=local_logs_dump_filepath)) as tar:
```

Comment:
Also, whichever archive type we use (and it should be zip so that windows users can also access it), I don't think we should test like this- let's instead stop all the cloudify services, then run the logs download both on the CLI VM and on the manager and check we get the same files.
Then, on the manager, we can check that the archive contains all the expected files by just comparing it to the target log files.

Reply:
And to update on this: The archive will indeed be different every time, so we can't test the way I suggested.